src/listener.js

```diff
@@ -40,7 +41,21 @@ module.exports = ({ handler, upgrader }, options = {}) => {
     const maConn = toConnection(channel, listener.__connections)
     log('new inbound connection %s', maConn.remoteAddr)
 
-    const conn = await upgrader.upgradeInbound(maConn)
+    channel.on('signal', (signal) => {
```

Comment:
This listener doesn't seem to be getting cleaned up anywhere. We need to make sure we're doing cleanup on the channel. I believe there is some code in the dialer that we should probably abstract out and reuse.

Reply:
I added the remove listeners. 

Regarding abstracting the dialer, I think the differences are significant enough to add complexity with abstraction. I tried some changes abstracting them, but I think it made things even more confusing. These event handlers with callbacks are always not clean 😖  I think I can change the dialer to removeAllListeners instead of providing the actual callback. With that, we can abstract a clean function that cleans all the listeners, but not sure if it is worth